cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -254,6 +254,12 @@ func GetProbeAddress(cfg *kubeadmapi.MasterConfiguration, componentName string)
 				}
 				// Return the IP if the URL contains an address instead of a name.
 				if ip := net.ParseIP(parsedURL.Hostname()); ip != nil {
+					// etcdctl doesn't support auto-converting zero addresses into loopback addresses
+					if ip.Equal(net.IPv4zero) || ip.Equal(net.IPv6zero) {
+						return "127.0.0.1"
```

Comment:
if we are in IPv6 configuration, better to use IPv6 loopback.

Reply:
This is intentional.
IPv6 endpoints don't work /w etcdctl.
I couldn't find an upstream bug -- perhaps we should file one.

Reply:
this is commented in the line below

Reply:
So, I tested this locally and using ::1 appears to work for etcdctl v3.1.12.

When testing, I had originally just modified the etcd manifest for a kubectl-based install, but neglected to update the certificate to add the ipv6 loopback address and was seeing this error: `Error:  grpc: timed out when dialing`. Once I updated the etcd server certificate to include ::1 I was able to use etcdctl with `--endpoints https://[::1]:2379` without an issue.

Reply:
@stealthybox We should probably update `GetEtcdAltNames()` in [pki_helpers.go](https://github.com/kubernetes/kubernetes/blob/16c5804482027393d532c47731a48c012878470d/cmd/kubeadm/app/phases/certs/pkiutil/pki_helpers.go) to add the ipv6 loopback address by default as well.

Reply:
@detiber Ah, that's totally the issue
I really struggled to determine what was wrong since it seems they removed the `--debug` flag

I'll fix this

Reply:
@stealthybox I ended up figuring it out by using etcdctl v3.2.x and using the `--debug` flag :rofl: